### PR TITLE
[5.3] Remove as on table alias

### PR DESCRIPTION
**Pull request by yajra**
This PR will:
- Remove ' as ' on table alias for self relationship. Fixes issue on yajra/laravel-oci8#211.
- Allow table aliasing without the optional as keyword for SQLITE.

A fix for PR #15776 and submitting to 5.3 as suggested in #15797

### CODE REVIEW

File: src/Illuminate/Database/Grammar.php [hunk 1]
```diff
@@ -63,6 +63,14 @@ public function wrap($value, $prefixAlias = false)
             }
 
             return $this->wrap($segments[0]).' as '.$this->wrapValue($segments[2]);
+        } elseif (strpos(strtolower($value), ' ') !== false) {
```
Comment (taylorotwell) [inline src/Illuminate/Database/Grammar.php]: This entire block needs more explanation.

Comment (yajra) [inline src/Illuminate/Database/Grammar.php]: @taylorotwell we first check if the table (from) value has `as` alias on first if condition. 
```sql
select u.* from users as u;
```

The second condition catches table alias declaration without `as` like:

```sql
select u.* from users u;
```
This catches the changes made on BelongsTo class, etc and fixes the unit tests.

Thanks!

File: src/Illuminate/Database/Grammar.php [hunk 2]
```diff
@@ -63,6 +63,14 @@ public function wrap($value, $prefixAlias = false)
...
+            $segments = explode(' ', $value);
+
+            if ($prefixAlias) {
+                $segments[1] = $this->tablePrefix.$segments[1];
+            }
+
+            return $this->wrap($segments[0]).' as '.$this->wrapValue($segments[1]);
         }
```
Comment (yajra) [inline src/Illuminate/Database/Grammar.php]: @taylorotwell sorry forgot to remove `as` on this line. Pushing another commit to fix this. Thanks for reviewing the PR.

**Comment by taylorotwell** [comment]
This is really something that should be handled in the OCI grammar you write custom. We don't support OCI out of the box. We support MySQL, Postgres, SQLite, and SQL Server. I think you could probably handle this in a custom wrap method for your OCI grammar.

**Comment by yajra** [comment]
@taylorotwell I think you have a point that this can be handled on my OCI Grammar. Didn't think of that. Will give it a shot and hope it works. Thanks!

**Pull request closed without merging**